westofengland-ca/weca_tdh_ckan
# Feature/tdh 351/ckan google analytics

**JIRA ticket: TDH-351**

**Change description and scope**
Added Google Analytics 4 to CKAN. It is disabled via a feature flag by default.
Configured with Cookie Control as optional.
Removed Cookie Control icon and added cookie settings link in the footer and policy page.

**How to test the change**
Currently deployed on Sandbox1.
GA4 wont track by default. Accept optional cookies in the Cookie Control settings and check that the GA tracking cookies now exist (e.g. `ga_`) on each page. Then revoke optional cookies and check that the GA cookies now don't exist.

**Has the code been formatted correctly (matches CKAN guidelines/PEP 8)?**
GA is integrated following the CIVIC Cookie Control recommended way. It uses the latest `gtag('consent', 'update')` functionality recommended by Google to revoke permissions. All tracking storage permissions are set to false by default.
